OsmAnd/src/net/osmand/plus/settings/fragments/configureitems/viewholders/PositionAnimationSettingCard.java

```diff
@@ -0,0 +1,132 @@
+package net.osmand.plus.settings.fragments.configureitems.viewholders;
+
+import static net.osmand.plus.utils.OsmAndFormatter.getFormattedPredictionTime;
+
+import android.view.View;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+
+import com.google.android.material.slider.Slider;
+
+import net.osmand.plus.OsmandApplication;
+import net.osmand.plus.R;
+import net.osmand.plus.helpers.AndroidUiHelper;
+import net.osmand.plus.settings.backend.preferences.CommonPreference;
+import net.osmand.plus.utils.ColorUtilities;
+import net.osmand.plus.utils.UiUtilities;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class PositionAnimationSettingCard {
+
+	private final OsmandApplication app;
+
+	private Integer[] range;
+	private int minValue;
+	private int maxValue;
+	private int currentValue;
+	private final CommonPreference<Integer> preference;
+
+	private final TextView title;
+	private final TextView summary;
+	private final TextView from;
+	private final TextView to;
+	private final View advancedItem;
+	private final ImageView advancedItemIcon;
+	private final View sliderContainer;
+	private final Slider slider;
+	private final boolean nightMode;
+
+	private boolean isSliderVisible;
+
+	public PositionAnimationSettingCard(@NonNull OsmandApplication app, @NonNull View view, @NonNull CommonPreference<Integer> preference, boolean nightMode) {
```

Comment:
If we have this card only for single specific preference, we can get it inside card, not pass as parameter. Also, app we could get from AndroidUtils.getApp(view.getContext)